src/adminjs.ts

```diff
@@ -157,12 +157,15 @@ class AdminJS {
     if (!Database || !Resource) {
       throw new Error('Adapter has to have both Database and Resource')
     }
+
+    // TODO: check if this is actually valid because "isAdapterFor" is always defined.
     // checking if both Database and Resource have at least isAdapterFor method
+    // @ts-ignore
     if (Database.isAdapterFor && Resource.isAdapterFor) {
```

Comment:
This line has been like this forever but I wonder if it's ever evaluated to false.

`BaseDatabase#isAdapterFor` and `BaseResource#isAdapterFor` are always defined and throw `NotImplementedError` so this check should always return true. We can theoretically make `BaseResource` and `BaseDatabase` abstract classes or we can simply remove this "if" condition but I decided to do it another time. I added `@ts-ignore` because Typescript v3 ignored this line while Typescript v4 says it always returns true.